website/statistics.py

```diff
@@ -340,6 +924,181 @@ def _calc_error_rate(fail, success):
     return (failed * 100) / max(1, failed + successful)
 
 
+def _determine_bool(bool_str):
+    if bool_str == "True":
+        return True
+    return False
+
+
+def _check_dashboard_display_args():
+    """
+    Checks the arguments of the request and returns the values. Mainly exists to avoid code duplication.
+    """
+    show_c1 = request.args.get("show_c1", default="True", type=str)
+    show_c1 = _determine_bool(show_c1)
+
+    show_c2 = request.args.get("show_c2", default="True", type=str)
+    show_c2 = _determine_bool(show_c2)
+
+    show_c3 = request.args.get("show_c3", default="True", type=str)
+    show_c3 = _determine_bool(show_c3)
+
+    student = request.args.get("student", default=None, type=str)
+    student = None if student == "None" else student
+
+    return show_c1, show_c2, show_c3, student
+
+
+def _get_available_adventures(adventures, teacher_adventures, customizations, last_adventures):
+    """
+    Returns the available adventures for all levels, given the possible adventures per level,
+    the teacher (adventures) and customization. Also adds how many students are currently in
+    progress for each adventure.
+
+    { level: [ { id, name, in_progress } ] }
+    """
+    teacher_adventures_formatted = {}
+    for adventure in teacher_adventures:
+        teacher_adventures_formatted[adventure['id']] = adventure["name"]
+
+    selected_adventures = {}
+    for level, adventure_list in customizations['sorted_adventures'].items():
+        adventures_for_level = []
+        for adventure in list(adventure_list):
+            adventure_key = adventure['name']
+
+            students_in_progress = []
+            for d in last_adventures[int(level) - 1]:
```

Comment:
I don't understand what the `d` variable level means, can you use a more descriptive variable name?